documentation/src/main/asciidoc/user_guide/clustering.adoc

```diff
@@ -943,27 +966,108 @@ If the application can handle losing some of the data in the cache, the
+utilising implementations of the <<merge-policies, EntryMergePolicy>> interface it is possible for said conflicts to
+be resolved automatically.
+
+[[merge-policies]]
+===== Merge Policies
+In the event of conflicts arising between one or more replicas of a given CacheEntry, it is necessary for a conflict
+resolution algorithm to be defined, therefore we provide the
+link:{javadocroot}/org/infinispan/conflict/EntryMergePolicy.html[EntryMergePolicy] interface. This interface consists
+of a single method, "merge", whose output is utilised as the "resolved" CacheEntry for a given key. A non-null return
+value is put to all replicas of the CacheEntry in question, whereas a null return value results in all replicas
+being removed from the cache.
+
+The merge method takes two parameters: the "preferredEntry" and "otherEntries". In the context of a partition merge,
+the preferredEntry is the CacheEntry associated with the partition whom's coordinator is conducting the merge (or if
+multiple entries exist in this partition, it's the primary replica). However, in all other contexts, the preferredEntry
+is simply the primary replica. The second parameter, "otherEntries" is simply a list of all other entries associated
+with the key for which a conflict was detected.
+
+NOTE: EntryMergePolicy::merge is only called when a conflict has been detected, it is not called if all CacheEntrys are
+the same.
+
+Currently Infinispan provides the following implementations of EntryMergePolicy:
+|===
+| Policy | Description
+
+| MergePolicies.PREFERRED_ALWAYS
+| Always utilise the "preferredEntry".
+
+| MergePolicies.PREFERRED_NON_NULL
+| Utilise the "preferredEntry" if it is non-null, otherwise utilise the first entry from "otherEntries".
+
+| MergePolicies.REMOVE_ALL
+| Always remove a key from the cache when a conflict is detected.
+|===
+
+==== Usage
+During a partition merge the ConflictManager automatically attempts to resolve conflicts utilising the configured
+EntryMergePolicy, however it is also possible to manually search for/resolve conflicts as required by your application.
+
+The code below shows how to retrieve an EmbeddedCacheManager's ConflictManager, how to retrieve all versions of a given
+key and how to check for conflicts across a given cache.
+
+[source,java]
+----
+EmbeddedCacheManager manager = new DefaultCacheManager("example-config.xml");
+Cache<Integer, String> cache = manager.getCache("testCache");
+ConflictManager<Integer, String> crm = ConflictManagerFactory.get(cache.getAdvancedCache());
+
+// Get All Versions of Key
+Map<Address, InternalCacheValue<String>> versions = crm.getAllVersions(1);
+
+// Process conflicts stream
+Stream<Map<Address, InternalCacheEntry<Integer, String>>> stream = crm.getConflicts();
+stream.forEach(System.out::println);
+
+// Detect and then resolve conflicts using the configured EntryMergePolicy
+crm.resolveConflicts();
```

Comment:
can we override the configured EntryMergePolicy? or it will always use the configured one.